activerecord/lib/active_record/insert_all.rb

```diff
@@ -0,0 +1,185 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class InsertAll
+    def initialize(model, inserts, **options)
+      raise ArgumentError, "inserts should be an array" unless inserts.is_a?(Array)
+      @model, @connection, @options = model, model.connection, options
+      @keys, @values_list = connection.prepare_keys_and_values_for_insert(inserts, model.table_name)
+    end
+
+    def execute
+      if values_list.empty?
```

Comment:
I'd flip this conditional around. Go with the common case first, which is that we have a values_list, go with the exception second. You can use #present? instead.

Reply:
👍